util/files/util.go

```diff
@@ -0,0 +1,64 @@
+package files
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+var RelativeOutOfBoundErr = errors.New("full path does not contain base path")
+
+// RelativePath will remove the basePath string from the fullPath
+// including the path separator. Differently from filepath.Rel, this
+// function will return error (RelativeOutOfBoundErr) if basePath
+// does not match (example 2).
+//
+// Example 1:
+//   fullPath: /home/test/app/readme.md
+//   basePath: /home/test
+//   return:   app/readme.md
+//
+// Example 2:
+//   fullPath: /home/test/app/readme.md
+//   basePath: /somewhere/else
+//   return:   "", RelativeOutOfBoundErr
+//
+// Example 3:
+//   fullPath: /home/test/app/readme.md
+//   basePath: /home/test/app/readme.md
+//   return:   .
+func RelativePath(fullPath, basePath string) (string, error) {
```

Comment:
All the tests assume two absolute paths. When this code is run as part of the CLI on a user's machine, the paths may be relative. We should either account for relative paths in tests, or change the CLI code to send absolute paths based on the current working directory.

I tracked down all the code paths to make sure there were no other users of this function which were using relative paths. Dumping that tree here.

* SendRepoStream
   * generateManifestsCmp
      * runConfigManagementPluginSidecars
         * GenerateManifests
            * getLocalObjectsString
               * getLocalObjects
                  * NewApplicationDiffCommand (appPath and repoPath are passed as CLI params)
               * NewApplicationSyncCommand (appPath and repoPath are passed as CLI params)
           * runManifestGenAsync (repoPath is a param, appPath comes from opContext)
              * runManifestGen (repoPath is a param, appPath comes from opContext)
                 * service.GenerateManifest (repoPath is a param, appPath comes from opContext)
                    * runRepoOperation (appPath comes either from `helmClient.ExtractChart` or `gitClient.Root()`)
                       In the case of Helm, repoPath is an absolute path, because it starts with the chart's temporary directory. appPath is the same as repoPath.
                       In the case of git, repoPath is an absolute path, because it's the root path of the git client which starts with os.TempDir(). appPath is repoPath with the path appended, so it's absolute.
   * matchRepositoryCMP (appPath and repoPath are the same)
      * DetectConfigManagementPlugin
         * runConfigManagementPluginSidecars (same paths as above)
         * Discover
            * ListApps (repoPath is gitClient.Root(), which is absolute)
            * AppType
               * GetAppSourceType
                  * GenerateManifests (same paths as above)
                * GetAppDetails (repoPath comes opContext)
                   * runRepoOperation (same paths as above)